app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,29 @@
+
+class SessionsController < ApplicationController
+
+  def new
+  end
+
+  def create
+    user = User.find_by(name: params[:session][:name])
+    if user && user.authenticate(params[:session][:password])
+      # puts " ++DEBUG++ into_login?"
+      current_user
+      log_in(user)
+      flash[:notice] = "Login successfully."
+      redirect_to user
+    else
+      # Create an error message
+      # puts " ++DEBUG++ login_fail?"
+      # puts "++DEBUG++ notice?"
+      redirect_to root_url, notice: 'Invalid username/password combination.'
+      # flash[:notice] = "Invalid username/password combination."
+    end
+  end
+
+  def destroy
+    log_out
+    redirect_to root_url, notice: 'Logged Out'
+  end
+
```

Comment:
So currently, we don't add the guest model right? @YEthYuan

Reply:
I think we can have a discussion on how to implement "unauthenticated user"